bin/templates/cordova/lib/build.js

```diff
@@ -422,11 +477,15 @@ function parseOpts(options, resolvedTarget) {
 module.exports.runClean = function(options) {
     var opts = parseOpts(options);
     var builder = builders[opts.buildMethod];
-    return builder.prepEnv()
+    return builder.prepEnv(opts)
     .then(function() {
-        return builder.clean(opts.extraArgs);
+        return builder.clean(opts);
     }).then(function() {
         shell.rm('-rf', path.join(ROOT, 'out'));
+    }).then(function() {
+        ['debug', 'release'].forEach(function(config) {
+            removeIfExists(path.join(ROOT, config + SIGNING_PROPERTIES));
```

Comment:
For non-CLI workflow, the user will likely create these files manually and be upset if clean deletes them. Probably best to just never delete them.